File: pre-contest/b_mayan_treasure/sol/b_mayan_treasure.php
<?php
fscanf(STDIN, "%d\n", $N);

$input = trim(fgets(STDIN));
$tab = explode(" ", $input);      
foreach($tab as &$number){
    $number = (int)$number;
}

sort($tab);

$a = 0;
$b = 0;
$c = 0;
$mini = -1;
$first = true;
for($i = 0; $i < $N - 2; $i++){
	if($first || $tab[$i + 2] - $tab[$i] < $mini){
		$first = false;
		$a = $tab[$i];
		$b = $tab[$i + 1];
		$c = $tab[$i + 2];
		$mini = $tab[$i + 2] - $tab[$i];
	}
}

echo($a . " " . $b . " " . $c);

?>

File: pre-contest/b_mayan_treasure/sol/b_mayan_treasure.py
#Taille de la liste
N = int(input())

#On récupère la liste
tab = [int(i) for i in input().split()]
#On trie la liste par ordre croissant
tab.sort()

#triplet de solutions a<=b<=c
a,b,c = 0,0,0
mini = -1
first = True
for i in range(N-2):
    if first or tab[i+2]-tab[i] < mini:
        first = False
        a,b,c = tab[i],tab[i+1],tab[i+2]
        mini = tab[i+2]-tab[i]
        
print(str(a)+" "+str(b)+" "+str(c))
